src/client/dfuse/dfuse_core.c

```diff
@@ -885,6 +889,47 @@ dfuse_ie_close(struct dfuse_projection_info *fs_handle,
 	D_FREE(ie);
 }
 
+#if HAVE_FUSE_LOG
+static void
+dfuse_fuse_log(enum fuse_log_level level,
+	const char *fmt,
+	va_list ap)
+{
+	int mask;
+	int d_level = DLOG_EMIT;
+
+	switch (level) {
+	case FUSE_LOG_EMERG:
+		d_level = DLOG_EMERG;
+		break;
+	case FUSE_LOG_ALERT:
+		d_level = DLOG_ALERT;
+		break;
+	case FUSE_LOG_CRIT:
+		d_level = DLOG_CRIT;
+		break;
+	case FUSE_LOG_ERR:
+		d_level = DLOG_ERR;
+		break;
+	case FUSE_LOG_WARNING:
+		d_level = DLOG_WARN;
+		break;
+	case FUSE_LOG_NOTICE:
+		d_level = DLOG_NOTE;
+		break;
+	case FUSE_LOG_INFO:
+		d_level = DLOG_INFO;
+		break;
+	case FUSE_LOG_DEBUG:
+		d_level = DLOG_DBG;
+		break;
+	}
+
+	mask = d_log_check(d_level | DB_IO | DD_FAC(fuse));
```

Comment:
maybe it doesn't matter much but you lose some efficiency of macros here.  The D_ macro creates a single variable that it checks (and caches the result) so you essentially bypass subsequent calls to d_log_check.

I think maybe what you want here is something like this:

int *flag = &DD_FLAG(DLOG_EMIT, D_LOGFAC);

with the switch also grabbing the right variable.

Then it's a matter of mimicking _D_DEBUG macro...so something like

if (__builtin_expect(*flag, 0)) {
     if (*flag == DLOG_UNINIT) {
          _D_LOG_CHECK(*flag, *flag | D_LOGFAC, ##__VA_ARGS__);
          break;
     }
_D_LOG_NOCHECK(*flag, ##__VA_ARGS__);

So maybe we need another macro here that takes a variable as an argument...and modify the existing calls to _D_DEBUG to call it with DD_FLAG result as an argument